modules/affile/logproto-file-writer.c

```diff
@@ -36,7 +31,7 @@
 typedef struct _LogProtoFileWriter
 {
   LogProtoClient super;
-  HTTPDestinationWorker *worker;
+  LogThreadedDestWorker *owner;
```

Comment:
We could store a `SignalSlotConnector *` pointer here, and it could be initialized by the constructor of LogProtoFileWriter: `log_proto_file_writer_new()`.

We need to pass the `AFFileDestDriver`'s `signal_slot_connector` instance to the constructor of LogProtoFileWriter.

Reply:
I think I'm doing this [here](https://imgur.com/a/9WXptpE)

Reply:
> I think I'm doing this [here](https://imgur.com/a/9WXptpE)
[commit-line](https://github.com/MrAnno/syslog-ng/blob/5f73c4f8922ce92afaa01817204f9fd35d8c0192/modules/affile/logproto-file-writer.c#L36)

Reply:
See my comment above. We need a specific instance from `AFFileDestDriver`, not a completely new one.

Also, please rename the field to `signal_slot_connector` in LogProtoFileWriter, as the connector is not the "owner" of the file writer.

Reply:
Updated :+1: